bin/test

```diff
@@ -1,35 +1,67 @@
 #!/usr/bin/env bash
 
-# Synopsis:
-# Test the track's exercises.
-# 
-# At a minimum, this file must check if the example/exemplar solution of each 
+# Test if the example/exemplar solution of each 
 # Practice/Concept Exercise passes the exercise's tests.
-#
-# To check this, you usually have to (temporarily) replace the exercise's solution files
-# with its exemplar/example files.
-#
-# If your track uses skipped tests, make sure to (temporarily) enable these tests
-# before running the tests.
-#
-# The path to the solution/example/exemplar files can be found in the exercise's
-# .meta/config.json file, or possibly inferred from the exercise's directory name.
 
 # Example:
 # ./bin/test
 
+set -eo pipefail
+
+docker pull exercism/unison-test-runner
+
+exit_code=0
+
+function run_test_runner() {
+    local dir=$1
+    local slug=$2
+
+    docker run \
+        --rm \
+        --network none \
+        --mount type=bind,src="${dir}",dst=/solution \
+        --mount type=bind,src="${dir}",dst=/output \
+        --tmpfs /tmp:rw \
+        exercism/unison-test-runner "${slug}" "/solution" "/output"
+}
+
+function verify_exercise() {
+    local dir=$(realpath $1)
+    local slug=$(basename "${dir}")
+    local implementation_file_key=$2    
+    local implementation_file=$(jq -r --arg d "${dir}" --arg k "${implementation_file_key}" '$d + "/" + .files[$k][0]' "${dir}/.meta/config.json")
+    local stub_file=$(jq -r --arg d "${dir}" '$d + "/" + .files.solution[0]' "${dir}/.meta/config.json")
```

Comment:
I've assumed that there is only ever one stub and exemplar/example file. If this does not hold, let me know and I'll update.